src/Ydb.Sdk/src/Services/Table/SessionPool.cs

```diff
@@ -147,7 +155,52 @@ public async Task<GetSessionResponse> GetSession()
                 _logger.LogWarning($"Failed to create session: {createSessionResponse.Status}");
             }
 
-            return new GetSessionResponse(createSessionResponse.Status);
+            ++_pendingSessions;
+
+            return null;
+        }
+
+        public async Task<GetSessionResponse> GetSession()
+        {
+            const int maxAttempts = 100;
+            var attempts = 0;
+
+            GetSessionResponse getSessionResponse = null!;
+            while (attempts < maxAttempts)
+            {
+                attempts++;
+                lock (_lock)
+                {
+                    var sesionResponse = TryGetFirstIdleSession();
+                    if (sesionResponse != null)
+                    {
+                        return sesionResponse;
+                    }
+
+                    sesionResponse = CheckSessionPoolLimit();
+                    if (sesionResponse != null)
+                    {
+                        return sesionResponse;
+                    }
+                }
+
+                var createSessionResponse = await _client.CreateSession(new CreateSessionSettings
+                {
+                    TransportTimeout = _config.CreateSessionTimeout,
+                    OperationTimeout = _config.CreateSessionTimeout
+                });
+
+                var sessionResponse = CheckCreateSessionStatus(createSessionResponse);
```

Comment:
Same problem - i don't understand (by function name) what returned in result: sucessfully result or error.